pkg/controller/apiextensions/composite/reconciler.go

```diff
@@ -257,6 +272,41 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 		"composition-name", comp.GetName(),
 	)
 
+	if meta.WasDeleted(cr) {
```

Comment:
Is there any chance we could detect when everything is at the same deletion priority and simply use Kubernetes garbage collection? I feel like that's the least surprising behaviour for users.